### ISSUE TITLE
Error during tests - build with NetCDF4 with PnetCDF already enabled

### ISSUE THREAD

**Issue by ArchangeGabriel** [Parallel-NetCDF/PnetCDF#33]
Compiling the recently release 1.11.0 test suite (with NetCDF support):
```
Making check in nf_test
make[2]: Entering directory '/build/pnetcdf-openmpi/src/pnetcdf-1.11.0/test/nf_test'
make  nf_test
make[3]: Entering directory '/build/pnetcdf-openmpi/src/pnetcdf-1.11.0/test/nf_test'
  PPF77    nf_error.o
m4  -DPNETCDF -I../../m4 test_get.m4 >test_get.F
  PPF77    test_write.o
  PPF77    util.o
  PPF77    nf_test.o
m4  -DPNETCDF -I../../m4 test_put.m4 >test_put.F
  PPF77    test_read.o
  CC       fortlib.o
m4  -DPNETCDF -I../../m4 test_iget.m4 >test_iget.F
m4  -DPNETCDF -I../../m4 test_iput.m4 >test_iput.F
  PPF77    test_get.o
  PPF77    test_put.o
  PPF77    test_iget.o
nf_test.F:477:71:

                             extra_flags=IOR(NF_NETCDF4,NF_CLASSIC_MODEL)
                                                                       1
Error: Symbol ‘nf_classic_model’ at (1) has no IMPLICIT type; did you mean ‘nf_ecmode’?
nf_test.F:477:54:

                             extra_flags=IOR(NF_NETCDF4,NF_CLASSIC_MODEL)
                                                      1
Error: Symbol ‘nf_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_ebadid’?
make[3]: *** [Makefile:737: nf_test.o] Error 1
make[3]: *** Waiting for unfinished jobs....
test_write.F:1763:47:

       if (nc_fmt .eq. NF_FORMAT_NETCDF4_CLASSIC) then
                                               1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_write.F:1848:39:

             version = NF_FORMAT_NETCDF4
                                       1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
make[3]: *** [Makefile:737: test_write.o] Error 1
test_put.F:2517:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:2518:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:2706:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:2707:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:2896:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:2897:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:3085:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:3086:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:3273:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:3274:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:3469:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:3470:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:3657:46:

             if ((format .EQ. NF_FORMAT_NETCDF4 .OR.
                                              1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:3658:54:

      +           format .EQ. NF_FORMAT_NETCDF4_CLASSIC) .AND.
                                                      1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:3899:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:3900:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:4140:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:4141:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:4382:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:4383:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:4623:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:4624:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:4863:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:4864:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:5111:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:5112:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:5351:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:5352:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:5598:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:5599:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:5880:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:5881:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:6163:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:6164:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:6445:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:6446:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:6726:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:6727:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:7015:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:7016:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:7296:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:7297:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:7579:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:7580:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:7869:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:7870:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:8160:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:8161:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:8450:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:8451:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:8739:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:8740:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:9036:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf5’?
test_put.F:9037:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
test_put.F:9325:45:

             if (format .EQ. NF_FORMAT_NETCDF4 .OR.
                                             1
Error: Symbol ‘nf_format_netcdf4’ at (1) has no IMPLICIT type; did you mean ‘nf_format_cdf2’?
test_put.F:9326:53:

      +          format .EQ. NF_FORMAT_NETCDF4_CLASSIC)
                                                     1
Error: Symbol ‘nf_format_netcdf4_classic’ at (1) has no IMPLICIT type; did you mean ‘nf_format_classic’?
make[3]: *** [Makefile:737: test_put.o] Error 1
make[3]: Leaving directory '/build/pnetcdf-openmpi/src/pnetcdf-1.11.0/test/nf_test'
make[2]: *** [Makefile:1014: check-am] Error 2
make[2]: Leaving directory '/build/pnetcdf-openmpi/src/pnetcdf-1.11.0/test/nf_test'
make[1]: *** [Makefile:481: check-recursive] Error 1
make[1]: Leaving directory '/build/pnetcdf-openmpi/src/pnetcdf-1.11.0/test'
make: *** [Makefile:514: check-recursive] Error 1
```

**Comment by ArchangeGabriel** [comment]
Also got one in parallel tests:
```
===========================================================
    examples/C: Parallel testing on 4 MPI processes
===========================================================
export SED="/usr/bin/sed"; export srcdir="."; export TESTOUTDIR="."; export TESTSEQRUN=""; export TESTMPIRUN="/usr/bin/mpiexec -n NP"; export PNETCDF_DEBUG="0"; export TESTPROGRAMS=""; export check_PROGRAMS="collective_write nonblocking_write nonblocking_write_in_def column_wise block_cyclic flexible_api get_info hints mput put_varn_float put_varn_int create_open global_attributes put_vara get_vara transpose transpose2D vard_int i_varn_int64 bput_varn_uint bput_varn_int64 fill_mode ghost_cell req_all vard_mvars time_var "; export ENABLE_BURST_BUFFER="0"; \
./parallel_run.sh 4 || exit 1
Error at collective_write.c:114: Feature is not enabled at configure time.
Error at collective_write.c:114: Feature is not enabled at configure time.
Error at collective_write.c:114: Feature is not enabled at configure time.
Error at collective_write.c:114: Feature is not enabled at configure time.
```

**Comment by ArchangeGabriel** [comment]
The first issue was due to compiling against a PnetCDF enabled NetCDF build. I’ve rebuilt NetCDF without PnetCDF support, then PnetCDF against this new NetCDF build, and it did not show this issue. I’m now rebuilding NetCDF against my new PnetCDF build. A bit cumbersome but not sure there is an easier way to do so.

BTW I’m wondering which way I should go in case of a future soname bump (because it seems that this is what triggered the issue here in the end): building PnetCDF without NetCDF support, then rebuilding NetCDF against the new soname, and finally rebuilding PnetCDF with NetCDF support, or the way I did it here (rebuild NetCDF without PnetCDF, then PnetCDF with NetCDF, and finally NetCDF with PnetCDF)?

The second error still remains though.

**Comment by ArchangeGabriel** [comment]
OK. Should I separate the Parallel test error in a dedicated issue?

**Comment by wkliao** [comment]
Yes, please.
That error is due to the assumption of option "--enable-profiling" is set at the configure time.

**Comment by wkliao** [comment]
Fixed in 48e8d21a2dece0b284354f0a405c547f32e9f304
Abort PnetCDF configure when detecting NetCDF4 was built with PnetCDF enabled

**Comment by ArchangeGabriel** [comment]
Thanks, awesome work fixing those issues so quickly. Apologies for not having been able to test the rc prior so that those could get detected before final release…
